src/pages/Lesson6.tsx

```diff
@@ -1,10 +1,74 @@
-import { FC } from 'react';
+import { FC, useState, useEffect } from 'react';
+import JustX from '../../public/Lesson6/Lesson6_JustX.png';
+import X from '../../public/Lesson6/Lesson6_X_arrow.png';
+import XY from '../../public/Lesson6/Lesson6_XY_arrow.png';
+import YX from '../../public/Lesson6/Lesson6_YX_arrow.png';
 import AppWrapper from '../components/AppWrapper';
 import CompareCode from '../components/CompareCode';
 import NavButtons from '../components/NavButtons';
 import Terminal from '../components/Terminal';
 import { HeaderSections } from '../types/globalTypes';
 import '../styles/PageWrapper.scss';
+import '../styles/CodeDiagram.scss';
+import '../styles/Terminal/Terminal.scss';
+
+export interface CodeDiagramProps {
+  code: {
+    code: string;
+  }[];
+}
+
+function CodeDiagram(props: CodeDiagramProps): JSX.Element {
+  const [clicked, setClick] = useState(Array(props.code.length).fill(false));
+  const [diagram, setDiagram] = useState(0);
+
+  useEffect(() => {
+    const newClicked = [...clicked]; // Create a shallow copy of the original array
+    newClicked[0] = true; // Modify the desired element
+    setClick(newClicked); // Update the state with the modified array
+  }, []);
+
+  return (
+    <div className="container">
+      <div className="codebox">
+        {props.code.map((code, index) => (
+          <div
+            className={clicked[index] ? 'clicked' : 'notclicked'}
+            onClick={() => {
+              const newClicked = [...clicked];
+              for (let i = 0; i < newClicked.length; i++) newClicked[i] = false;
+              newClicked[index] = true;
+              setClick(newClicked);
+              setDiagram(index);
+            }}
+            key={index}
+          >
+            <p>{code.code}</p>
+          </div>
+        ))}
+      </div>
+      <div style={{ height: '402px', paddingTop: '1.5em' }}>
+        {diagram == 0 ? (
```

Comment:
Yeah, I ran into that problem before...not exactly sure what causes it though. Also, why did you copy the component to the lesson 6 page? We should just keep it in the [existing component](https://github.com/uclaacm/parcel-pointers/blob/main/src/components/CodeDiagram.tsx). We probably want to do the same exact thing with lesson 5 [here](https://github.com/uclaacm/parcel-pointers/issues/353)
Let's do something like we did with the slideshow instead, where we just pass an array of images as props and then index the array instead:
https://github.com/uclaacm/parcel-pointers/blob/main/src/pages/PointerMotivation.tsx

Reply:
Oh I see now. Actually, I don't think you have to do that because we won't be using the component with FillArrows, I think we will only be using it with images since its only used in lesson 5 and 6:
<img width="874" alt="image" src="https://github.com/uclaacm/parcel-pointers/assets/78634083/33da6e6a-9175-46c0-a5c5-24239f11aacb">